miscellaneous/etroc1_cal_studies.py

```diff
@@ -0,0 +1,214 @@
+from pathlib import Path # Pathlib documentation, very useful if unfamiliar:
+                         #   https://docs.python.org/3/library/pathlib.html
+
+import lip_pps_run_manager as RM
+
+import logging
+import pandas
+import numpy
+import sqlite3
+import plotly.graph_objects as go
+
+from utilities import apply_event_filter, filter_dataframe
+
+def plot_cal_code_task(
+    Fermat: RM.RunManager,
+    script_logger: logging.Logger,
+    task_name:str,
+    data_file:Path,
+    filter_files:dict[str,Path] = {},
+    drop_old_data:bool=True,
+    extra_title: str = "",
+    full_html:bool=False,
+    ):
+
+    with Fermat.handle_task(task_name, drop_old_data=drop_old_data) as Monet:
+        base_path=Monet.task_path
+        run_name=Monet.run_name,
+        with sqlite3.connect(data_file) as input_sqlite3_connection:
+            data_df = pandas.read_sql('SELECT * FROM etroc1_data', input_sqlite3_connection, index_col=None)
+
+            data_df = filter_dataframe(
+                df=data_df,
+                filter_files=filter_files,
+                script_logger=script_logger,
+            )
+
+            filter_df = pandas.read_feather(Monet.path_directory/"event_filter.fd")
+            filter_df.set_index("event", inplace=True)
+
+            from cut_etroc1_single_run import apply_event_filter
+            data_df = apply_event_filter(data_df, filter_df)
+            df = data_df.loc[data_df['accepted']==True]
+
+            # Even - Even
+            fig = go.Figure()
+            for board_id in sorted(df["data_board_id"].unique()):
+                condition = (df["data_board_id"] == board_id) & (df["time_of_arrival"]%2 == 0) & (df["calibration_code"]%2 == 0)
+                fig.add_trace(go.Histogram(
+                    x=df.loc[condition]["calibration_code"],
+                    name='Board {}'.format(board_id), # name used in legend and hover labels
+                    opacity=0.5,
+                    bingroup=1,
+                ))
+            fig.update_layout(
+                barmode='overlay',
+                title_text="Histogram of Even Calibration Code when Even TOA Code<br><sup>Run: {}{}</sup>".format(run_name, extra_title),
+                xaxis_title_text='Calibration Code', # xaxis label
+                yaxis_title_text='Count', # yaxis label
+            )
+            fig.update_yaxes(type="log")
+
+            fig.write_html(
+                base_path/'calibration_code_histogram_evenTOA_evenCAL.html',
+                full_html = full_html,
+                include_plotlyjs = 'cdn',
+            )
+
+            # Even - Odd
```

Comment:
If you factorize the three plots below, it can be done into a single function where you place two parameters, one for the parity of TOA and the other for the parity of calcode.